src/incentives/ccip_bridge.sol

```diff
@@ -0,0 +1,190 @@
+
+    // Event emitted when a message is sent to another chain.
+    event BridgeRequest(
+        bytes32 indexed messageId,
+        uint64 indexed destinationChainSelector,
+        address sender,
+        uint256 id 
+    );
+
+    // Event emitted when a message is received from another chain.
+    event BridgeReciept(
+        bytes32 indexed messageId,
+        uint64 indexed sourceChainSelector,
+        address receiver,
+        uint256 id
+    );
+
+    modifier onlyWhitelistedDestinationChain(uint64 _destinationChainSelector) {
+        if (!whitelistedDestinationChains[_destinationChainSelector])
+            revert DestinationChainNotWhitelisted(_destinationChainSelector);
+        _;
+    }
+
+    modifier onlyWhitelistedSourceChain(uint64 _sourceChainSelector) {
+        if (!whitelistedSourceChains[_sourceChainSelector])
+            revert SourceChainNotWhitelisted(_sourceChainSelector);
+        _;
+    }
+
+    modifier onlyWhitelistedSenders(address _sender) {
+        if (!whitelistedSenders[_sender]) revert SenderNotWhitelisted(_sender);
+        _;
+    }
+
+    modifier canBridge() {
+        require(!paused, "Bridging is paused");
+        _;
+    }
+
+    constructor(address router) CCIPReceiver(router) {
+        // Polygon
+        whitelistedDestinationChains[4051577828743386545] = true;
+        whitelistedSourceChains[4051577828743386545] = true;
+
+        // ETH
+        whitelistedDestinationChains[5009297550715157269] = true;
+        whitelistedSourceChains[5009297550715157269] = true;
+    }
+
+    fallback() external payable { }
+    receive() external payable { }
+
+    function setAffinePassAddress(address _affinePass) external onlyOwner {
+        affinePass = IAffinePass(_affinePass);
+    }
+
+    function setPaused(bool _paused) external onlyOwner {
+        paused = _paused;
+    }
+
+    function setChainReciever(uint64 _chainId, address _router) external onlyOwner {
+        chainReciever[_chainId] = _router;
+    }
+
+    function bridgePass(
+        uint64 destinationChainSelector,
+        address receiver,
+        uint256 id
+    ) external onlyWhitelistedDestinationChain(destinationChainSelector) canBridge returns (bytes32 messageId) {
+        require(affinePass.ownerOf(id) == msg.sender, "Not owner of token");
+
+        Client.EVM2AnyMessage memory message = _buildCCIPMessage(
+            chainReciever[destinationChainSelector],
+            receiver,
+            id,
+            address(0)
+        );
+
+        uint256 fee = IRouterClient(i_router).getFee(
+            destinationChainSelector,
+            message
+        );
+
+        if (fee > address(this).balance)
+            revert NotEnoughBalance(address(this).balance, fee);
+
+        messageId = IRouterClient(i_router).ccipSend{value: fee}(
+            destinationChainSelector,
+            message
+        );
```

Comment:
If the check necessary? It seems like `ccipSend()` will revert if the fee is greater than the balance of the contract.